x/sequencers/types/msgs.go

```diff
@@ -0,0 +1,174 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+var (
+	_ sdk.Msg                            = (*MsgCreateSequencer)(nil)
+	_ sdk.Msg                            = (*MsgUpdateSequencer)(nil)
+	_ codectypes.UnpackInterfacesMessage = (*MsgCreateSequencer)(nil)
+)
+
+func (m *MsgCreateSequencer) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	return unpacker.UnpackAny(m.PubKey, new(cryptotypes.PubKey))
+}
+
+func (m *MsgCreateSequencer) ValidateBasic() error {
+	if _, err := m.OperatorAccAddr(); err != nil {
+		return errorsmod.Wrap(errors.Join(gerrc.ErrInvalidArgument, err), "acc addr")
+	}
+	if m.GetPubKey() == nil {
+		return errors.New("pub key is nil")
+	}
+	if m.GetPubKey().GetCachedValue() == nil {
+		return errors.New("pub key cached value is nil")
+	}
+	v := stakingtypes.Validator{
+		ConsensusPubkey: m.GetPubKey(),
+	}
+	tm, err := v.TmConsPublicKey()
+	if err != nil {
+		return errorsmod.Wrap(err, "tm cons pub key")
+	}
+	if tm.GetEd25519() == nil {
+		return errors.New("not ed5519")
+	}
+	operator, err := m.OperatorAddr()
+	if err != nil {
+		return errorsmod.Wrap(errors.Join(gerrc.ErrInvalidArgument, err), "acc addr")
+	}
+	pubKey, _ := v.ConsPubKey()
+
+	// We return OK only if the key and sig contains a key and signature where the signature was produced by the key, and the signature
+	// is over the operator account, and the app payload data (if not empty).
+	//
+	// The reasoning is as follows:
+	//
+	// We know from the SDK TX signing mechanism that the account originates from the operator, and on this chain ID.
+	// Therefore, we just need to check that the consensus private key also signed over this account and chain ID.
```

Comment:
I blv this godoc is wrong as we're only checking if signed on validator address.